### PR TITLE
Use context manager for file handling in some system modules.

### PR DESCRIPTION
**Pull request by atombrella**
##### SUMMARY
Small chunk from #63840 

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
system, core.

### CODE REVIEW

**Comment by atombrella** [comment]
ready_for_review

File: lib/ansible/modules/system/crypttab.py [hunk 1]
```diff
@@ -173,14 +170,12 @@ def __init__(self, path):
         if not os.path.exists(path):
             if not os.path.exists(os.path.dirname(path)):
                 os.makedirs(os.path.dirname(path))
-            open(path, 'a').close()
+        with open(path, 'a'):
```
Comment (felixfontein) [inline lib/ansible/modules/system/crypttab.py]: This needs to be indented, to make sure the file is only opened for appending when it doesn't yet exist.

**Review by felixfontein — COMMENTED** [review]
All other changes look good (and safe).

File: lib/ansible/modules/system/crypttab.py (hunk 1 above)
Comment (atombrella) [reply lib/ansible/modules/system/crypttab.py]: Great catch :face_with_head_bandage: Sorry. I'll change it. Weird why all tests pass with this, though.

Comment (felixfontein) [reply lib/ansible/modules/system/crypttab.py]: It doesn't destroy anything, but it touches the file unnecessarily. That's why the tests don't complain I guess.

Comment (atombrella) [reply lib/ansible/modules/system/crypttab.py]: I updated it, however I'm actually not sure if the line of code does anything ...

**Review by felixfontein — COMMENTED** [review]
Changes look good. A changelog fragment would be good. (Probably better to mention the individual modules which are changed instead of some general category.)

**Comment by Akasurde** [comment]
@atombrella Can you please add a changelog entry as suggested by @felixfontein? Thanks.

**Comment by atombrella** [comment]
@Akasurde Yes. Sorry for the delay, I kind of overlooked/forgot the request about the changelog.

**Review by felixfontein — APPROVED** [review]

**Comment by atombrella** [comment]
@felixfontein Any reason why this has not been merged? A changelog fragment was added as requested, and CI is happy.

**Comment by Akasurde** [comment]
@atombrella Could you please rebase this PR? `cronvar`, `crypttab` and `parted` moved to [`community.general`](https://github.com/ansible-collections/community.general) so you might want to raise a PR against this repo.

needs_info

**Comment by atombrella** [comment]
> @atombrella Could you please rebase this PR? `cronvar`, `crypttab` and `parted` moved to [`community.general`](https://github.com/ansible-collections/community.general) so you might want to raise a PR against this repo.

Yes. I'll rebase. Completely forgot about the PR. You guys are not particularly fast in merging trivial stuff.

**Comment by bcoca** [comment]
@atombrella most if it is due to python version compatibility issues, since we now only support 2.7 we can finally merge code that doesn't work with 2.6 (for example,  compound contexts). Also this is not really fixing a bug, but altering style (to more modern python).

**Comment by atombrella** [comment]
@bcoca With some delay, this has rebased.

**Pull request merged**